tests/unittests/test_tests_catalog.py

```diff
@@ -0,0 +1,431 @@
+import csv
+import pytest
+from vane import test_catalog_client
+
+
+# Global test parameters
+TEST_DIR = "sample_network_tests/api/"
+test_directories = ["sample_network_tests/api/"]
+
+
+# pylint: disable=redefined-outer-name
+@pytest.fixture
+def loginfo(mocker):
+    """
+    Fixture to mock logger info calls from vane.test_catalog_client
+    """
+    return mocker.patch("vane.vane_logging.logging.info")
+
+
+@pytest.fixture
+def logdebug(mocker):
+    """
+    Fixture to mock logger debug calls from vane.test_catalog_client
+    """
+    return mocker.patch("vane.vane_logging.logging.debug")
+
+
+def file_clean_up(file_name):
+    """Clean up files created in a test
+
+    Args:
+        file_name(str): File name to delete
+    """
+
+    if os.path.isfile(file_name):
+        os.remove(file_name)
+
+
+def test_test_catalog_client_constructor(loginfo, logdebug):
+    """
+    Unit Test for TestCatlogClient object __init__ method
+    """
+
+    _ = test_catalog_client.TestCatalogClient(TEST_DIR)
+
+    loginfo_calls = [
+        call("Creating Test catalog Client object"),
+    ]
+    loginfo.assert_has_calls(loginfo_calls, any_order=False)
+
+    logdebug_calls = [
+        call(f"Set Test catalog Client object directory to {TEST_DIR}"),
+    ]
+    logdebug.assert_has_calls(logdebug_calls, any_order=False)
+
+
+def test_write_test_catalog(loginfo, mocker):
+    """
+    Unit Test for TestCatlogClient object, write_test_catalog method
+    """
+
+    mocker_object = mocker.patch("vane.test_catalog_client.TestCatalogClient.walk_dir")
+    test_catalog = test_catalog_client.TestCatalogClient(TEST_DIR)
+    test_catalog.write_test_catalog()
+    mocker_object.assert_called_once()
+
+    loginfo_calls = [
+        call("Start writing test catalog"),
+        call("Ending writing test catalog file"),
+    ]
+    loginfo.assert_has_calls(loginfo_calls, any_order=False)
+
+
+def test_walk_dir(logdebug, mocker):
+    """
+    Unit Test for TestCatlogClient object, method walk_dir
+    """
+    mocker_object = mocker.patch("vane.test_catalog_client.TestCatalogClient.get_parsed_data")
+    test_catalog = test_catalog_client.TestCatalogClient([TEST_DIR])
+    test_catalog.walk_dir()
+    mocker_object.assert_called_once()
+
+    all_files = [["__init__.py", "test_api.py", "test_definition.yaml"]]
+    discovered_file = {
+        "sample_network_tests/api/": [
+            "sample_network_tests/api/test_api.py",
+            "sample_network_tests/api/test_definition.yaml",
+        ]
+    }
+
+    logdebug_calls = [f"Walking directory {TEST_DIR} for test cases"]
+
+    # As test catalog collects files recursively added zip
+    for test_directory, files in zip(test_directories, all_files):
+        logdebug_calls.extend([f"Discovered files {files} in directory {test_directory}"])
+
+    logdebug_calls.append(f"Discovered test files: {discovered_file} for parsing")
+
+    for debug_msg in logdebug_calls:
+        logdebug.assert_any_call(debug_msg)
```

Comment:
we dont need the for loop here
```suggestion
       logdebug.assert_has_calls(logdebug_calls, any_order=False)
```